### PR TITLE
SPOTAUT-12351: SDK Implementation for AWS Beanstalk APIs

### PR DESCRIPTION
**Pull request by ramrutha497**
Checklist
---------

JIRA Ticket -

Checklist (Please mark Yes/No/NA for each of the questions below) -

- I have provided clear and detailed explanation of changes made in the proposed PR : Yes
- I have compiled the code successfully my laptop : Yes
- I have written examples to demonstrates the usage of the newly implemented APIs : Yes
- I have incremented the release version : No
- I have run the Intellij Code Analysis on newly added code and fixed all (valid) reported issues : No

OR

- I have taken approval from scrum master or project manager for skipping any of the items above : No

### CODE REVIEW

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupService.java [hunk 1]
```diff
@@ -2106,4 +2106,191 @@ public static ApiElastigroupGetInstanceStatusResponse getInstanceStatus(String i
...
+        ElastigroupApiResponse instanceStatusResponse = getCastedResponse(response, ElastigroupApiResponse.class);
+
+        if (instanceStatusResponse.getResponse().getCount() > 0) {
+            retVal = instanceStatusResponse.getResponse().getItems().get(0);
+        }
+
+        return retVal;
+    }
+
+    public static Boolean startBeanstalkMaintenance(String groupId,
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupService.java:2223]: format of code required

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupService.java [hunk 2]
```diff
@@ -2106,4 +2106,191 @@ public static ApiElastigroupGetInstanceStatusResponse getInstanceStatus(String i
...
+        }
+
+        // Get the headers for AWS.
+        Map<String, String> headers = buildHeaders(authToken);
+
+        // Build URI
+        String uri = String.format("%s/aws/ec2/group/%s/beanstalk/reimport", apiEndpoint, groupId);
+
+        // Send the request.
+        RestResponse response = RestClient.sendGet(uri, headers, queryParams);
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupService.java:2211]: this should be put call

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java [hunk 3]
```diff
@@ -1036,4 +1036,94 @@ public RepoGenericResponse<ElastigroupGetInstanceStatusResponse> getInstanceStat
...
+
+        try {
+            ApiElastigroup apigetBeanstalkConfig = SpotinstElastigroupService.getBeanstalkConfig(environmentId, region, authToken, account);
+            Elastigroup    instanceStatus    = ElastigroupConverter.toBl(apigetBeanstalkConfig);
+            retVal = new RepoGenericResponse<>(instanceStatus);
+        }
+        catch (SpotinstHttpException e) {
+            retVal = ExceptionHelper.handleHttpException(e);
+        }
+
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java:1052]: remove extra line space

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java [hunk 4]
```diff
@@ -1036,4 +1036,94 @@ public RepoGenericResponse<ElastigroupGetInstanceStatusResponse> getInstanceStat
...
+        try {
+            ApiElastigroupGetBeanstalkMaintenanceStatusResponse apigetBeanstalkMaintenanceStatus = SpotinstElastigroupService.getBeanstalkMaintenanceStatus(groupId, authToken, account);
+            ElastigroupGetBeanstalkMaintenanceStatusResponse    instanceStatus    = ElastigroupConverter.toBl(apigetBeanstalkMaintenanceStatus);
+            retVal = new RepoGenericResponse<>(instanceStatus);
+        }
+        catch (SpotinstHttpException e) {
+            retVal = ExceptionHelper.handleHttpException(e);
+        }
+
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java:1068]: remove extra line space

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java [hunk 5]
```diff
@@ -1036,4 +1036,94 @@ public RepoGenericResponse<ElastigroupGetInstanceStatusResponse> getInstanceStat
...
+        return retVal;
+    }
+
+    @Override
+    public RepoGenericResponse<Boolean> startBeanstalkMaintenance(String groupId, String authToken, String account) {
+        RepoGenericResponse<Boolean> retVal;
+
+        try {
+
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java:1093]: remove extra line spaces

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java [hunk 6]
```diff
@@ -1036,4 +1036,94 @@ public RepoGenericResponse<ElastigroupGetInstanceStatusResponse> getInstanceStat
...
+        return retVal;
+
+    }
+
+    @Override
+    public RepoGenericResponse<Boolean> finishBeanstalkMaintenance(String groupId, String authToken, String account) {
+        RepoGenericResponse<Boolean> retVal;
+
+        try {
+
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java:1114]: remove extra line spaces

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java [hunk 7]
```diff
@@ -1036,4 +1036,94 @@ public RepoGenericResponse<ElastigroupGetInstanceStatusResponse> getInstanceStat
...
+
+            Boolean finishBeanstalkMaintenanceResponse = SpotinstElastigroupService
+                    .finishBeanstalkMaintenance(groupId, authToken, account);
+
+            retVal = new RepoGenericResponse<>(finishBeanstalkMaintenanceResponse);
+        }
+
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java:1121]: remove extra line space

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java [hunk 8]
```diff
@@ -1036,4 +1036,94 @@ public RepoGenericResponse<ElastigroupGetInstanceStatusResponse> getInstanceStat
 
         return retVal;
     }
+
+    @Override
+    public RepoGenericResponse<Elastigroup> getBeanstalkConfig(String environmentId, String region, String authToken, String account) {
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java:1041]: re-arrange the functions as is in openApi document

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupClient.java [hunk 9]
```diff
@@ -1314,4 +1314,108 @@ public ElastigroupGetInstanceStatusResponse getInstanceStatus(String instanceId)
         return retVal;
     }
 
+    public Elastigroup getBeanstalkConfig(String environmentId, String region) {
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupClient.java:1317]: re-arrange the functions as is in openApi document

File: src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java [hunk 10]
```diff
@@ -203,6 +204,25 @@ public static void main(String[] args) throws IOException {
         System.out.println("----------Get Instance Status--------------");
         ElastigroupGetInstanceStatusResponse response = getInstanceStatus(elastigroupClient, instanceId);
 
+        // Get Beanstalk Configuration
+        System.out.println("----------Get Beanstalk Configuration--------------");
+        String getBeanstalkId = getBeanstalkConfig(elastigroupClient, "environmentId", "region");
+
+        // Start Beanstalk Maintenance
+        System.out.println("----------Start Beanstalk Maintenance--------------");
+        Boolean startBeanstalkMaintenance = startBeanstalkMaintenance(elastigroupClient, getBeanstalkId);
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java:213]: the parameter should be groupId

File: src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java [hunk 11]
```diff
@@ -203,6 +204,25 @@ public static void main(String[] args) throws IOException {
...
+
+        // Get Beanstalk Maintenance Status
+        System.out.println("----------Get Beanstalk Maintenance Status--------------");
+        String getBeanstalkMaintenanceStatus = getBeanstalkMaintenanceStatus(elastigroupClient, getBeanstalkId);
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java:217]: the parameter should be groupId

File: src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java [hunk 12]
```diff
@@ -203,6 +204,25 @@ public static void main(String[] args) throws IOException {
...
+
+        // Finish Beanstalk Maintenance
+        System.out.println("----------Finish Beanstalk Maintenance--------------");
+        Boolean finishBeanstalkMaintenance = finishBeanstalkMaintenance(elastigroupClient, getBeanstalkId, getBeanstalkMaintenanceStatus);
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java:221]: the parameter should be groupId

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupService.java [hunk 13]
```diff
@@ -2106,4 +2106,191 @@ public static ApiElastigroupGetInstanceStatusResponse getInstanceStatus(String i
...
+        Map<String, String> headers = buildHeaders(authToken);
+
+        // Build URI
+        String uri = String.format("%s/aws/ec2/group/beanstalk/import", apiEndpoint);
+
+        // Send the request.
+        RestResponse response = RestClient.sendGet(uri, headers, queryParams);
+
+        // Handle the response.
+        ElastigroupApiResponse getBeanstalkConfigResponse = getCastedResponse(response, ElastigroupApiResponse.class);
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupService.java:2144]: create elasticBeanstalk class (BL and API) to catch the response

**Review by chandra1-n — CHANGES_REQUESTED** [review]
please fix the review comments

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupService.java [hunk 14]
```diff
@@ -2106,4 +2106,191 @@ public static ApiElastigroupGetInstanceStatusResponse getInstanceStatus(String i
 
         return retVal;
     }
+
+    public static ApiElastigroup getBeanstalkConfig(String environmentId, String region, String authToken,
```
[the remark this answers is not in the document]
Comment (ramrutha497) [reply src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupService.java:2110]: Done

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupService.java (hunk 1 above)
Comment (ramrutha497) [reply src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupService.java:2223]: Done

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupService.java (hunk 2 above)
Comment (ramrutha497) [reply src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupService.java:2211]: Done

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupService.java (hunk 1 above)
Comment (ramrutha497) [reply src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupService.java:2223]: Done

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java (hunk 3 above)
Comment (ramrutha497) [reply src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java:1052]: Done

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java (hunk 4 above)
Comment (ramrutha497) [reply src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java:1068]: Done

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java (hunk 5 above)
Comment (ramrutha497) [reply src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java:1093]: Done

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java (hunk 6 above)
Comment (ramrutha497) [reply src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java:1114]: Done

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java (hunk 7 above)
Comment (ramrutha497) [reply src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java:1121]: Done

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java (hunk 8 above)
Comment (ramrutha497) [reply src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java:1041]: Done

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupClient.java (hunk 9 above)
Comment (ramrutha497) [reply src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupClient.java:1317]: Done

File: src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java (hunk 10 above)
Comment (ramrutha497) [reply src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java:213]: Done

File: src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java (hunk 11 above)
Comment (ramrutha497) [reply src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java:217]: Done

File: src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java (hunk 12 above)
Comment (ramrutha497) [reply src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java:221]: Done

File: src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java [hunk 15]
```diff
@@ -1231,4 +1251,66 @@ private static ElastigroupGetInstanceStatusResponse getInstanceStatus(SpotinstEl
...
+        ElastigroupGetBeanstalkMaintenanceStatusResponse response = client.getBeanstalkMaintenanceStatus(groupId);
+
+        System.out.println(String.format("Beanstalk maintenance status: %s", response.getStatus()));
+
+        return response.getStatus();
+    }
+
+    private static Boolean finishBeanstalkMaintenance(SpotinstElastigroupClient client, String groupId, String currentStatus) {
+
+        System.out.println(String.format("Finish beanstalk maintenance. groupId: %s", groupId));
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java:1296]: change the print statement to "Finish beanstalk maintenance of the group: %s with current status: %s"

File: src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java [hunk 16]
```diff
@@ -1231,4 +1251,66 @@ private static ElastigroupGetInstanceStatusResponse getInstanceStatus(SpotinstEl
...
+        System.out.println(String.format("Finish beanstalk maintenance. currentStatus: %s", currentStatus));
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java:1297]: this print statement can be removed

File: src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java [hunk 17]
```diff
@@ -1231,4 +1251,66 @@ private static ElastigroupGetInstanceStatusResponse getInstanceStatus(SpotinstEl
...
+        Boolean response = false;
+        if(currentStatus.equals("AWAIT_USER_UPDATE"))
+            response = client.finishBeanstalkMaintenance(groupId);
+
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java:1301]: remove extra line space

File: src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java [hunk 18]
```diff
@@ -1231,4 +1251,66 @@ private static ElastigroupGetInstanceStatusResponse getInstanceStatus(SpotinstEl
...
+        System.out.println(String.format("Finish beanstalk maintenance response: %s", response));
+
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java:1303]: remove extra line space

File: src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java [hunk 19]
```diff
@@ -1231,4 +1251,66 @@ private static ElastigroupGetInstanceStatusResponse getInstanceStatus(SpotinstEl
...
+        return response;
+    }
+
+    private static Elastigroup beanstalkReimport(SpotinstElastigroupClient client, String groupId) {
+
+        System.out.println(String.format("Beanstalk reimport. groupId: %s", groupId));
+        Elastigroup response = client.beanstalkReimport(groupId);
+
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java:1311]: remove extra line space

File: src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java [hunk 20]
```diff
@@ -1231,4 +1251,66 @@ private static ElastigroupGetInstanceStatusResponse getInstanceStatus(SpotinstEl
...
+        System.out.println(String.format("Instance ID: %s", response.getId()));
+
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java:1313]: remove extra line space

File: src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java [hunk 21]
```diff
@@ -1231,4 +1251,66 @@ private static ElastigroupGetInstanceStatusResponse getInstanceStatus(SpotinstEl
...
+            response = client.finishBeanstalkMaintenance(groupId);
+
+        System.out.println(String.format("Finish beanstalk maintenance response: %s", response));
+
+        return response;
+    }
+
+    private static Elastigroup beanstalkReimport(SpotinstElastigroupClient client, String groupId) {
+
+        System.out.println(String.format("Beanstalk reimport. groupId: %s", groupId));
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java:1309]: change the statement as mentioned above

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java [hunk 22]
```diff
@@ -165,6 +169,125 @@ private static ApiCodeDeploy toDal(ElastigroupCodeDeploy codeDeploy) {
...
+                retVal.setGracePeriod(deploymentPreferences.getGracePeriod());
+            }
+
+            if (deploymentPreferences.isStrategySet()) {
+                if (deploymentPreferences.getStrategy() != null) {
+                    retVal.setStrategy(toDal(deploymentPreferences.getStrategy()));
+                }
+            }
+        }
+
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java:223]: remove extra line space

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java [hunk 23]
```diff
@@ -165,6 +169,125 @@ private static ApiCodeDeploy toDal(ElastigroupCodeDeploy codeDeploy) {
...
+        return retVal;
+
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java:225]: remove extra line space

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java [hunk 24]
```diff
@@ -165,6 +169,125 @@ private static ApiCodeDeploy toDal(ElastigroupCodeDeploy codeDeploy) {
...
+                }
+            }
+
+            if (beanstalk.isManagedActionsSet()) {
+                if (beanstalk.getManagedActions() != null) {
+                    retVal.setManagedActions(toDal(beanstalk.getManagedActions()));
+                }
+            }
+        }
+
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java:194]: remove extra line space

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java [hunk 25]
```diff
@@ -165,6 +169,125 @@ private static ApiCodeDeploy toDal(ElastigroupCodeDeploy codeDeploy) {
...
+        return retVal;
+
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java:196]: remove extra line space

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java [hunk 26]
```diff
@@ -165,6 +169,125 @@ private static ApiCodeDeploy toDal(ElastigroupCodeDeploy codeDeploy) {
...
+            if (strategy.isActionSet()) {
+                retVal.setAction(strategy.getAction());
+            }
+
+            if (strategy.isShouldDrainInstancesSet()) {
+                retVal.setShouldDrainInstances(strategy.getShouldDrainInstances());
+            }
+        }
+
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java:242]: remove extra line space

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java [hunk 27]
```diff
@@ -165,6 +169,125 @@ private static ApiCodeDeploy toDal(ElastigroupCodeDeploy codeDeploy) {
...
+        return retVal;
+
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java:244]: remove extra line space

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java [hunk 28]
```diff
@@ -165,6 +169,125 @@ private static ApiCodeDeploy toDal(ElastigroupCodeDeploy codeDeploy) {
...
+        if (managedActions != null) {
+            retVal = new ApiManagedActions();
+
+            if (managedActions.isPlatformUpdateSet()) {
+                if (managedActions.getPlatformUpdate() != null) {
+                    retVal.setPlatformUpdate(toDal(managedActions.getPlatformUpdate()));
+                }
+            }
+        }
+
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java:259]: remove extra line space, across all similar places

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java [hunk 29]
```diff
@@ -165,6 +169,125 @@ private static ApiCodeDeploy toDal(ElastigroupCodeDeploy codeDeploy) {
...
+        return retVal;
+
+    }
+
+    private static ApiBeanstalkPlatformUpdate toDal(BeanstalkPlatformUpdate strategy) {
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java:264]: change variable name to platformUpdate

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java [hunk 30]
```diff
@@ -1754,6 +1881,131 @@ private static ElastigroupCodeDeploy toBl(ApiCodeDeploy apiCodeDeploy) {
...
+                }
+            }
+            blManagedActions = blManagedActionsBuilder.build();
+        }
+
+        return blManagedActions;
+
+    }
+
+    private static BeanstalkPlatformUpdate toBl(ApiBeanstalkPlatformUpdate apiStrategy) {
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java:1980]: change variable name to apiPlatformUpdate

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java [hunk 31]
```diff
@@ -3451,4 +3703,19 @@ public static ElastigroupGetInstanceStatusResponse toBl(
         return retVal;
 
     }
+
+    public static ElastigroupGetBeanstalkMaintenanceStatusResponse toBl(
+            ApiElastigroupGetBeanstalkMaintenanceStatusResponse elastigroupGetBeanstalkStatusResponse) {
+        ElastigroupGetBeanstalkMaintenanceStatusResponse retVal = null;
+
+        if (elastigroupGetBeanstalkStatusResponse != null) {
+            retVal = new ElastigroupGetBeanstalkMaintenanceStatusResponse();
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java:3712]: use builder class, as utilized by other methods

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupService.java (hunk 13 above)
Comment (ramrutha497) [reply src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupService.java]: Done

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupClient.java [hunk 32]
```diff
@@ -1314,4 +1314,108 @@ public ElastigroupGetInstanceStatusResponse getInstanceStatus(String instanceId)
...
+        }
+        else {
+            List<HttpError> httpExceptions = startBeanstalkMaintenanceResponse.getHttpExceptions();
+            HttpError       httpException  = httpExceptions.get(0);
+            LOGGER.error(
+                    String.format("Error encountered while attempting to start beanstalk maintenance. Code: %s. Message: %s.",
+                            httpException.getCode(), httpException.getMessage()));
+            throw new SpotinstHttpException(httpException.getMessage());
+        }
+
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupClient.java:1334]: remove extra line space wherever is applied

File: src/main/java/com/spotinst/sdkjava/model/api/elastigroup/aws/ApiBeanstalkPlatformUpdate.java [hunk 33]
```diff
@@ -0,0 +1,105 @@
...
+        isSet.add("updateLevel");
+        this.updateLevel = updateLevel;
+    }
+
+    //endregion
+
+    //region isSet methods
+    // Is instanceRefreshEnabled Set boolean method
+    @JsonIgnore
+    public boolean isInstanceRefreshEnabledSet() {
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/api/elastigroup/aws/ApiBeanstalkPlatformUpdate.java:82]: use java Boolean data type

File: src/main/java/com/spotinst/sdkjava/model/bl/elastigroup/aws/BeanstalkPlatformUpdate.java [hunk 34]
```diff
@@ -0,0 +1,135 @@
+package com.spotinst.sdkjava.model.bl.elastigroup.aws;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import java.util.HashSet;
+import java.util.Set;
+
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/bl/elastigroup/aws/BeanstalkPlatformUpdate.java:7]: user ignore=true and NON_NULL in all bl and api calsses

**Review by chandra1-n — CHANGES_REQUESTED** [review]
please fix the comments

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java (hunk 29 above)
Comment (ramrutha497) [reply src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java:264]: Done

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java (hunk 30 above)
Comment (ramrutha497) [reply src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java:1980]: Done

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java (hunk 31 above)
Comment (ramrutha497) [reply src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java:3712]: Done

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupClient.java (hunk 32 above)
Comment (ramrutha497) [reply src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupClient.java:1334]: Done

File: src/main/java/com/spotinst/sdkjava/model/bl/elastigroup/aws/BeanstalkPlatformUpdate.java (hunk 34 above)
Comment (ramrutha497) [reply src/main/java/com/spotinst/sdkjava/model/bl/elastigroup/aws/BeanstalkPlatformUpdate.java:7]: Done

**Review by anuragsharma-123 — APPROVED** [review]

**Review by chandra1-n — APPROVED** [review]

**Pull request merged**